fiat-core/src/main/java/com/netflix/spinnaker/fiat/model/resources/ResourceType.java

```diff
@@ -16,32 +16,60 @@
 
 package com.netflix.spinnaker.fiat.model.resources;
 
-import javax.annotation.Nonnull;
+import java.util.Objects;
 
-public enum ResourceType {
-  ACCOUNT(Account.class),
-  APPLICATION(Application.class),
-  SERVICE_ACCOUNT(ServiceAccount.class), // Fiat service account.
-  ROLE(Role.class),
-  BUILD_SERVICE(BuildService.class);
+public class ResourceType {
 
-  public Class<? extends Resource> modelClass;
+  private final String name;
 
-  ResourceType(Class<? extends Resource> modelClass) {
-    this.modelClass = modelClass;
+  public static ResourceType ACCOUNT = new ResourceType("account");
+  public static ResourceType APPLICATION = new ResourceType("application");
+  public static ResourceType SERVICE_ACCOUNT =
+      new ResourceType("service_account"); // Fiat service account.
+  public static ResourceType ROLE = new ResourceType("role");
+  public static ResourceType BUILD_SERVICE = new ResourceType("build_service");
+
+  public ResourceType(String name) {
+    this.name = name;
   }
 
   // TODO(ttomsu): This is Redis-specific, so it probably shouldn't go here.
-  public static ResourceType parse(@Nonnull String pluralOrKey) {
+  public static ResourceType parse(String pluralOrKey) {
+    if (pluralOrKey == null) {
```

Comment:
This is explicitly tested for. Not sure why, but didn't want to change anything.

Reply:
hmm, I think previously or at some point the @Nonnull was lombok's annotation rather than javax.annotation that causes it to throw on a null argument

Reply:
Yeah I wanted to switch to the lombok annotation, but the exception type for `@NonNull` is configured in `lombok.config` is `IllegalArgumentException`.